### PR TITLE
[CS0618] `StreamRefsSpec` `Sink.ActorRef<TIn>(IActorRef, object)` is obsolete

### PR DESCRIPTION
**Pull request by eaba**
## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [x] This change follows the [Akka.NET API Compatibility Guidelines](https://getakka.net/community/contributing/api-changes-compatibility.html).
* [x] This change follows the [Akka.NET Wire Compatibility Guidelines](https://getakka.net/community/contributing/wire-compatibility.html).

### CODE REVIEW

File: src/core/Akka.Streams.Tests/Dsl/StreamRefsSpec.cs [hunk 1]
```diff
@@ -94,7 +94,7 @@ protected override bool Receive(object message)
                          * We write out code, knowing that the other side will stream the data into it.
                          * For them it's a Sink; for us it's a Source.
                          */
-                        var sink = StreamRefs.SinkRef<string>().To(Sink.ActorRef<string>(_probe, "<COMPLETE>"))
+                        var sink = StreamRefs.SinkRef<string>().To(Sink.ActorRef<string>(_probe, "<COMPLETE>", ex => new Status.Failure(ex)))
```
Comment (eaba) [inline src/core/Akka.Streams.Tests/Dsl/StreamRefsSpec.cs:97]: Changed into `Sink<TIn, NotUsed> ActorRef<TIn>(IActorRef actorRef, object onCompleteMessage, Func<Exception, object> onFailureMessage)`

File: src/core/Akka.Streams.Tests/Dsl/StreamRefsSpec.cs [hunk 2]
```diff
@@ -109,7 +109,7 @@ protected override bool Receive(object message)
                     {
                         var sink = StreamRefs.SinkRef<string>()
                             .WithAttributes(StreamRefAttributes.CreateSubscriptionTimeout(TimeSpan.FromMilliseconds(500)))
-                            .To(Sink.ActorRef<string>(_probe, "<COMPLETE>"))
+                            .To(Sink.ActorRef<string>(_probe, "<COMPLETE>", ex => new Status.Failure(ex)))
```
Comment (eaba) [inline src/core/Akka.Streams.Tests/Dsl/StreamRefsSpec.cs:112]: Changed into `Sink<TIn, NotUsed> ActorRef<TIn>(IActorRef actorRef, object onCompleteMessage, Func<Exception, object> onFailureMessage)`

File: src/core/Akka.Streams.Tests/Dsl/StreamRefsSpec.cs [hunk 3]
```diff
@@ -242,7 +242,7 @@ public void SourceRef_must_send_messages_via_remoting()
             _remoteActor.Tell("give");
             var sourceRef = ExpectMsg<ISourceRef<string>>();
 
-            sourceRef.Source.RunWith(Sink.ActorRef<string>(_probe.Ref, "<COMPLETE>"), Materializer);
+            sourceRef.Source.RunWith(Sink.ActorRef<string>(_probe.Ref, "<COMPLETE>", ex => new Status.Failure(ex)), Materializer);
```
Comment (eaba) [inline src/core/Akka.Streams.Tests/Dsl/StreamRefsSpec.cs:245]: Changed into `Sink<TIn, NotUsed> ActorRef<TIn>(IActorRef actorRef, object onCompleteMessage, Func<Exception, object> onFailureMessage)`

File: src/core/Akka.Streams.Tests/Dsl/StreamRefsSpec.cs [hunk 4]
```diff
@@ -255,7 +255,7 @@ public void SourceRef_must_fail_when_remote_source_failed()
             _remoteActor.Tell("give-fail");
             var sourceRef = ExpectMsg<ISourceRef<string>>();
 
-            sourceRef.Source.RunWith(Sink.ActorRef<string>(_probe.Ref, "<COMPLETE>"), Materializer);
+            sourceRef.Source.RunWith(Sink.ActorRef<string>(_probe.Ref, "<COMPLETE>", ex => new Status.Failure(ex)), Materializer);
```
Comment (eaba) [inline src/core/Akka.Streams.Tests/Dsl/StreamRefsSpec.cs:258]: Changed into `Sink<TIn, NotUsed> ActorRef<TIn>(IActorRef actorRef, object onCompleteMessage, Func<Exception, object> onFailureMessage)`

File: src/core/Akka.Streams.Tests/Dsl/StreamRefsSpec.cs [hunk 5]
```diff
@@ -269,7 +269,7 @@ public void SourceRef_must_complete_properly_when_remote_source_is_empty()
             _remoteActor.Tell("give-complete-asap");
             var sourceRef = ExpectMsg<ISourceRef<string>>();
 
-            sourceRef.Source.RunWith(Sink.ActorRef<string>(_probe.Ref, "<COMPLETE>"), Materializer);
+            sourceRef.Source.RunWith(Sink.ActorRef<string>(_probe.Ref, "<COMPLETE>", ex => new Status.Failure(ex)), Materializer);
```
Comment (eaba) [inline src/core/Akka.Streams.Tests/Dsl/StreamRefsSpec.cs:272]: Changed into `Sink<TIn, NotUsed> ActorRef<TIn>(IActorRef actorRef, object onCompleteMessage, Func<Exception, object> onFailureMessage)`

**Review by eaba — COMMENTED** [review]
Changes

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**